### PR TITLE
fix(app-checks): change type aliases to allow for normal funcs

### PR DESCRIPTION
**Pull request by ooliver1**
## Summary

Fixes the application checks to allow for normal `async def` functions before they are decorated as a command.
Also allows for cog versions of the above.

(adding `CoroFunc` to the callable union)

the other partially related change is so app checks can be used on cogs w/o type errors

## Checklist

- [x] If code changes were made then they have been tested.
  - [x] I have updated the documentation to reflect the changes.
  - [x] I have run `task pyright` and fixed the relevant issues.
- [ ] This PR fixes an issue.
- [ ] This PR adds something new (e.g. new method or parameters).
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: nextcord/types/checks.py [hunk 1]
```diff
@@ -34,7 +34,9 @@
 Coro = Coroutine[Any, Any, T]
 MaybeCoro = Union[T, Coro[T]]
 CoroFunc = Callable[..., Coro[Any]]
-ApplicationCheck = Callable[[Interaction], MaybeCoro[bool]]
+ApplicationCheck = Union[
+    Callable[[ClientCog, Interaction], bool], Callable[[Interaction], MaybeCoro[bool]]
```
Comment (DenverCoder1) [inline nextcord/types/checks.py:38]: Should the first bool also be in `MaybeCoro`?

Comment (ooliver1) [inline nextcord/types/checks.py:38]:
```suggestion
    Callable[[ClientCog, Interaction], MaybeCoro[bool]], Callable[[Interaction], MaybeCoro[bool]]
```

**Review by DenverCoder1 — APPROVED** [review]
Looks good, I haven't tested

File: nextcord/types/checks.py [hunk 2]
```diff
@@ -34,7 +34,9 @@
 Coro = Coroutine[Any, Any, T]
 MaybeCoro = Union[T, Coro[T]]
 CoroFunc = Callable[..., Coro[Any]]
-ApplicationCheck = Callable[[Interaction], MaybeCoro[bool]]
+ApplicationCheck = Union[
+    Callable[[ClientCog, Interaction], MaybeCoro[bool]], Callable[[Interaction], MaybeCoro[bool]]
```
Comment (Skelmis) [inline nextcord/types/checks.py:38]: Do these nested brackets not require a Union?

Comment (ooliver1) [reply nextcord/types/checks.py:38]: They represent the arguments to the function

Comment (Skelmis) [reply nextcord/types/checks.py:38]: In that case, shouldnt it be `List[..]`?

**Review by Skelmis — APPROVED** [review]

**Pull request merged**